parsers/mcmaster.js

```diff
@@ -1,283 +1,146 @@
-		classtime[1] = classtime[1].replace(AM_PM,"");
-		stminute = classtime[0].split(":")[1];
-		etminute = classtime[1].split(":")[1];
-
-		date = Date.today().set({year:start_date[0],month:start_date[1],day:start_date[2]}) //.parse(dayString.toLowerCase())
-		if (dayString === "Monday"){
-			date = date.monday()
-		}
-		if (dayString === "Tuesday"){
-			date = date.tuesday()
-		}
-		if (dayString === "Wednesday"){
-			date = date.wed()
-		}
-		if (dayString === "Thursday"){
-			date = date.thursday()
-		}
-		if (dayString === "Friday"){
-			date = date.fri()
-		}
-		if (dayString === "Saturday"){
-			date = date.sat()
-		}
-		if (dayString === "Sunday") {
-			date = date.sunday()
-		}
-		times[0]=  [date.getFullYear(),date.getMonth(),date.getDate(), hour[0] , parseInt(stminute)];
-		times[1] = [date.getFullYear(),date.getMonth(),date.getDate(), hour[1] , parseInt(etminute)];
-		return times;
+/**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate
+ * @param {String} endDate
+ * @return {String Array} [startDate, endDate]  with format MM/DD/YYYY or YYYY/MM/DD or YYYY-MM-DD
+ */
+function fixDateFormat(sem_start, sem_end) {
+    var startDateSplit = sem_start.split("/");
+    var endDateSplit = sem_end.split("/");
+
+    var startMonth = parseInt(startDateSplit[0], 10);
+    var startDay = parseInt(startDateSplit[1], 10);
+    var startYear = parseInt(startDateSplit[2], 10);
+
+    var endMonth = parseInt(endDateSplit[0], 10);
+    var endDay = parseInt(endDateSplit[1], 10);
+    var endYear = parseInt(endDateSplit[2], 10);
+
+    var dayDiff = endDay - startDay;
+    // If it contains the '-' or the first slot is YYYY it is already in correct format
+    if (sem_start.indexOf('-') > -1 || startMonth.toString().length === 4) {
+        return [sem_start, sem_end]
+    }
+    // Date is DD/MM/YYYY as month can't be > 12, switch the MM and DD or If the difference is 3 months(Fall/Winter), 1 month(Spring/Summer) or -5 months(multi-semester(Sept-Apr)),
+    else if (startMonth > 12 || endMonth > 12 || (startYear === endYear && startMonth > endMonth) ||
+    ([1,3,-5].indexOf(dayDiff) > -1 && [1, 5, 7, 9].indexOf(startDay) > -1)) {
+        return [[startDay, startMonth, startYear].join("/"),[endDay, endMonth, endYear].join("/")];
+    }
+
+    return [sem_start, sem_end]
 }
 
-var convertToCal = function(filePath, start_date,end_date, directory, filename, callback) {
-	var converted = convert(filePath);
-	return calParser(converted[0], converted[1], start_date,end_date, directory, filename, callback);
+/**
+ * Takes in the raw data of timetable and get all the important details including, course name, professor, times etc.
+ * @param {String} text
+ * @return {Object/JSON} timetable
+ */
+function parse(text){
+    var _wsRegex = /\s+/;
```

Comment:
these regex variablies should be global variables (outside this function and at the top of the file after requires)